cli/demo_teeracle_generic.sh

```diff
@@ -0,0 +1,132 @@
+#!/bin/bash
+set -euo pipefail
+
+trap "echo The demo is terminated (SIGINT); exit 1" SIGINT
+trap "echo The demo is terminated (SIGTERM); exit 1" SIGTERM
+
+# Registers a teeracle with the parentchain, and publish some oracle data.
+#
+# Demo to show that an enclave can update the exchange rate only when
+#   1. the enclave is registered at the pallet-teerex.
+#   2. and that the code used is reliable -> the enclave has been put the teeracle whitelist via a governance or sudo
+#   call.
+#
+# The teeracle's whitelist has to be empty at the start. So the script needs to run with a clean node state.
+# A registered mrenclave will be added in the whitelist by a sudo account. Here //Alice
+#
+# setup:
+# run all on localhost:
+#   integritee-node purge-chain --dev
+#   integritee-node --dev -lpallet_teeracle=debug,parity_ws=error,aura=error,sc_basic_authorship=error
+#   integritee-service --clean-reset run (--skip-ra --dev)
+#
+# then run this script
+#
+# usage:
+#   demo_teeracle_generic.sh -p <NODEPORT> -P <WORKERPORT> -d <DURATION> -i <WORKER_UPDATE_INTERVAL> -u <NODE_URL> -V <WORKER_URL> -C <CLIENT_BINARY_PATH>
+
+while getopts ":p:P:d:i:u:V:C:" opt; do
+    case $opt in
+        p)
+            NPORT=$OPTARG
+            ;;
+        P)
+            WORKER1PORT=$OPTARG
+            ;;
+        d)
+            DURATION=$OPTARG
+            ;;
+        i)
+            INTERVAL=$OPTARG
+            ;;
+        u)
+            NODEURL=$OPTARG
+            ;;
+        V)
+            WORKER1URL=$OPTARG
+            ;;
+        C)
+            CLIENT_BIN=$OPTARG
+            ;;
+    esac
+done
+
+# using default port if none given as arguments
+NPORT=${NPORT:-9944}
+NODEURL=${NODEURL:-"ws://127.0.0.1"}
+
+WORKER1PORT=${WORKER1PORT:-2000}
+WORKER1URL=${WORKER1URL:-"wss://127.0.0.1"}
+
+CLIENT_BIN=${CLIENT_BIN:-"./../bin/integritee-cli"}
+
+DURATION=${DURATION:-48}
+INTERVAL=${INTERVAL:-86400}
+
+LISTEN_TO_ORACLE_EVENTS_CMD="oracle listen-to-oracle-events"
+ADD_TO_WHITELIST_CMD="oracle add-to-whitelist"
+
+echo "Using client binary ${CLIENT_BIN}"
+echo "Using node uri ${NODEURL}:${NPORT}"
+echo "Using trusted-worker uri ${WORKER1URL}:${WORKER1PORT}"
+echo "Using worker data update interval ${INTERVAL}"
+echo "Count the update events for ${DURATION}"
+echo ""
+
+OPEN_METEO="https://api.open-meteo.com/"
+let "MIN_EXPECTED_NUM_OF_EVENTS=$DURATION/$INTERVAL-3"
```

Comment:
Haha, I see that there was already a magic number before.

Reply:
One magic number exchanged for another :)